src/content/en/2021/caching.md

```diff
@@ -17,4 +19,328 @@ featured_stat_3: TODO
+
+The adoption of Service Workers has continued to steadily increase. While 1% of Mobile pages registered a service worker in 2020, 9% of Mobile pages ranked in the top 1,000 registered one in 2021.
+
+{{ figure_markup(
+  image="sw-adoption.png",
+  caption="Service Worker adoption.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=688062455&format=interactive",
+  sheets_gid="802609299",
+  sql_file="service_worker_rank.sql"
+) }}
+
+The primary way to cache resources within a Service Worker, it's by using the CacheStorage API. This allows a developer to create a custom cache strategy for any requests passing through the worker; some famous ones are Stale-While-Revalidate, Cache Falling Back to Network, Network Falling Back to Cache, Cache Only. In recent years become even easier to adopt those strategies by increasing the popularity of [Workbox](https://developers.google.com/web/tools/workbox/modules/workbox-strategies), which decides what cache you want to plug and play.
+
+## Headers adoption
+
+With both CDN and Browser, the HTTP Headers are the leading toolkit each developer needs to master to properly cache resources. Headers are simply instructions known as "directives", read during the HTTP Request or Response to control the cache strategy.
+
+The caching-related headers, or the absence of them, tell the browser or CDN three essential pieces of information:
+- **Cacheability**: Is this content cacheable?
+- **Freshness**: If it is cacheable, how long can it be cached for? `Cache-Control` and `Expires` 
+- **Validation**: If it is cacheable, how do I ensure that my cached version is still fresh?
+
+Headers are meant to be used either alone or together. To tell if the content is cachable and fresh, we have:
+- `Expires` specifies an explicit expiration date and time (i.e., when precisely the content expires)
+- `Cache-Control` specifies a cache duration (i.e., how long the content can be cached in the browser relative to when it was requested)
+In case both are specified, `Cache-Control` takes precedence.
+
+The usage of the Cache-Control header has increased steadily since 2019. 74% of responses on Mobile pages included the Cache-Control header, while 75% of responses on Desktop pages utilized the header. From 2020, the usage of this specific header increased by 0.71% for Mobile and by 1.13% for Desktop.
+
+{{ figure_markup(
+  image="cache-control-expires.png",
+  caption="Percent of responses that set `Cache-Control` and `Expires` headers.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=816405026&format=interactive",
+  sheets_gid="2102749619",
+  sql_file="header_trends.sql"
+) }}
+
+To validate the content, we have:
```

Comment:
Edits: 

To validate the content, we have:

`Last-Modified` indicates when the object was last changed. Its value is a date timestamp.
- `ETag` (Entity Tag) provides a unique identifier for the content as a quoted string. It can take any format the server chooses; it is typically a hash of the file contents, but it could be a timestamp or a simple string.

When both are specified, `ETag` takes precedence.